plugins/inputs/kafka_consumer/kafka_consumer.go

```diff
@@ -170,6 +176,32 @@ func (k *KafkaConsumer) startErrorAdder(acc telegraf.Accumulator) {
 func (k *KafkaConsumer) Start(acc telegraf.Accumulator) error {
 	var err error
 
+	// If TopicRegexps is set, add matches to Topics
+	if len(k.TopicRegexps) > 0 {
+		k.Log.Infof("adding topic regexps %+v", k.TopicRegexps)
+		err = k.addTopicRegexps()
+		if err != nil {
+			return err
+		}
+		// If, additionally, TopicRefreshInterval is set,
+		// schedule the refresh.
+		if k.TopicRefreshInterval != 0 {
+			k.Log.Infof("refreshing topics every %f s",
+				float64(k.TopicRefreshInterval)/float64(1e9))
+			tick := time.Duration(k.TopicRefreshInterval)
+			ticker := time.NewTicker(tick)
+			go func() {
+				for range ticker.C {
+					k.Log.Infof("refreshing topics")
+					err = k.addTopicRegexps()
+					if err != nil {
+						k.Log.Errorf("topic refresh failed: %v", err)
+					}
```

Comment:
Again, please limit the scope to make unintended (reuse) effects more unlikely
```suggestion
					if err := k.addTopicRegexps(); err != nil {
						k.Log.Errorf("topic refresh failed: %v", err)
					}
```